client/components/Archive/Archive.tsx

```diff
@@ -38,6 +48,27 @@ export default class Archive extends Component<{}, ArchiveState> {
     }
   }
 
+  // helper function which buffers the current state's list
+  // state list is displayed in InifiniteScroll component
+  loadMoreItems = (ac: ArchiveContainer, cuc: CurrentUserContainer) => {
+    ac.retrieveCompanies(PAGE_LENGTH, this.state.skip).then((pagination) => {
```

Comment:
Will these retrieve repeats?

Reply:
The combination of the number of items to skip and the number of items to retrieve make sure we don't retrieve the same company twice. Is this what you're asking?